tests/user/update-my-dog-registration.test.ts

```diff
@@ -0,0 +1,198 @@
+import { updateMyDogRegistration } from "@/lib/user/actions/update-my-dog-registration";
+import { withDb } from "../_db_helpers";
+import {
+  getDogMapper,
+  getUserActor,
+  insertCall,
+  insertDog,
+  insertReport,
+  insertUser,
+  insertVet,
+} from "../_fixtures";
+import { MyDogRegistrationUpdate } from "@/lib/user/user-models";
+import { UTC_DATE_OPTION, parseDateTime } from "@/lib/utilities/bark-time";
+import {
+  CALL_OUTCOME,
+  DogAntigenPresence,
+  DogGender,
+  PARTICIPATION_STATUS,
+  YesNoUnknown,
+} from "@/lib/data/db-enums";
+import { dbInsertDogVetPreference } from "@/lib/data/db-dogs";
+import { Pool, PoolClient } from "pg";
+import { dbBegin, dbQuery, dbRelease } from "@/lib/data/db-utils";
+
+describe("updateMyDogRegistration", () => {
+  it("should return OK_UPDATED when update was successful", async () => {
+    await withDb(async (dbPool) => {
+      // GIVEN users u1 with dog d1
+      const u1 = await insertUser(1, dbPool);
+      const d1 = await insertDog(1, u1.userId, dbPool);
+
+      // WHEN
+      const actor1 = getUserActor(dbPool, u1.userId);
+      const update = registrationUpdate(d1.dogId);
```

Comment:
This update does not contain any preferred vet. The test would cover more ground if (i) in the GIVEN scenario the preferred vet was v1 and in the update the preferred vet is v2.

Reply:
Updated